science/veryfasttree/Portfile

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+
+github.setup        citius veryfasttree 4.0.3 v
+github.tarball_from archive
+revision            0
+
+categories          science
+license             GPL-3
+platforms           darwin
+maintainers         usc.es:cesaralfredo.pineiro
+description         Efficient phylogenetic tree inference for massive taxonomic datasets
+long_description    {*}${description}
+homepage            https://github.com/citiususc/veryfasttree
+
+checksums           rmd160  a4070bf0ddc4aa0448f9a7878a3587a9250197a0 \
+                    sha256  770e41773806c65e60ebc77363beccf40e3d339affd97c07eb7b2e1acb51af60 \
+                    size    898112
+
+depends_lib         port:libomp
+
+configure.args-append -DUSE_NATIVE=OFF
+configure.cxxflags-append "-I${prefix}/include/libomp -L${prefix}/lib/libomp"
+
+variant AVX description {Add AVX extensions} {
+    if {${build_arch} eq "i386" || ${build_arch} eq "x86_64"} {
+        configure.args-append -DUSE_AVX=ON
+    }
+}
+
+variant AVX2 description {Add AVX2 extensions} {
+    if {${build_arch} eq "i386" || ${build_arch} eq "x86_64"} {
+        configure.args-append -DUSE_AVX2=ON
+    }
+}
+
+variant AVX512 description {Add AVX extensions} {
+    if {${build_arch} eq "i386" || ${build_arch} eq "x86_64"} {
+        configure.args-append -DUSE_AVX512=ON
+    }
+}
```

Comment:
are these options listed in the variants here detected at configure/runtime? If so, then the usual way would be to add an `native` variant and just set all of the options to ON. You should be able to `grep` for `variant native` and find other ports doing the same (often, `OpenMP` is involved in such ports as well so perhaps you can check some of the syntax for enabling that in this port as well).

Reply:
Yes good point, I replaced all variants with a native.